nav2_costmap_2d/plugins/inflation_layer.cpp

```diff
@@ -407,4 +417,64 @@ InflationLayer::generateIntegerDistances()
   return level;
 }
 
+/**
+  * @brief Callback executed when a parameter change is detected
+  * @param event ParameterEvent message
+  */
+rcl_interfaces::msg::SetParametersResult
+InflationLayer::dynamicParametersCallback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+  std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());
+  rcl_interfaces::msg::SetParametersResult result;
+
+  bool need_cache_recompute = false;
+
+  for (auto parameter : parameters) {
+    const auto & param_type = parameter.get_type();
+    const auto & param_name = parameter.get_name();
+
+    if (param_type == ParameterType::PARAMETER_DOUBLE) {
+      if (param_name == name_ + "." + "inflation_radius" &&
+        inflation_radius_ != parameter.as_double())
+      {
+        inflation_radius_ = parameter.as_double();
+        cell_inflation_radius_ = cellDistance(inflation_radius_);
+        need_reinflation_ = true;
+        need_cache_recompute = true;
+      } else if (param_name == name_ + "." + "cost_scaling_factor" &&
+        cost_scaling_factor_ != parameter.as_double())
+      {
+        cost_scaling_factor_ = parameter.as_double();
+        need_reinflation_ = true;
+        need_cache_recompute = true;
+      }
+    } else if (param_type == ParameterType::PARAMETER_BOOL) {
+      if (param_name == name_ + "." + "enabled" && enabled_ != parameter.as_bool()) {
+        enabled_ = parameter.as_bool();
+        need_reinflation_ = true;
+        current_ = false;
+      } else if (param_name == name_ + "." + "inflate_unknown" &&
+        inflate_unknown_ != parameter.as_bool())
+      {
+        inflate_unknown_ = parameter.as_bool();
+        need_reinflation_ = true;
+      } else if (param_name == name_ + "." + "inflate_around_unknown" &&
+        inflate_around_unknown_ != parameter.as_bool())
+      {
+        inflate_around_unknown_ = parameter.as_bool();
+        need_reinflation_ = true;
+      }
+    }
+
+  }
+
+  if (need_cache_recompute) {
```

Comment:
Wouldn't it be better at this point to call `matchSize` which will handle `computeCaches()` and do all the `cellDistance()` stuff for you? That way the if/else conditions above can be more simple.

Reply:
I guess you are right, the decision of computeCache() or matchSize() I just made based on the ROS1 Nav stack https://github.com/ros-planning/navigation/blob/noetic-devel/costmap_2d/plugins/inflation_layer.cpp

Practically I think the only difference between the two approaches is that through matchSize() we are additionally resetting the resolution_ and seen vectors. But it doesnt seem like that causes any issues. Pushing the change.